### DISCUSSION TITLE
Some things that I personally think can be improved

### DISCUSSION THREAD

**Discussion by LucasZhanye** [prest/prest#882] [API Server]
When I use prestd,I think there are the following three points that can be improved：
1. Configuration dynamics

3. Unified response structure,such as
```
{
  "code": 404,
  "reason": "",
 "message": "pq: relation "public.hell" does not exist",
"metadata": {}
}
```
4. Table permissions
I think access to all tables should be allowed by default. Only tables with permissions set in the configuration will perform permission control. In this way, combined with dynamic configuration, the permissions of the tables can be controlled without restarting the application.

Welcome to discuss, if possible, I hope to submit a PR

**Comment by avelino** [comment]
@LucasZhanye I have some doubts about the points you are suggesting for improvements

> Configuration dynamics

can you give us an example of what dynamic configuration would be?

> Table permissions

Are you proposing that access permissions to the table be configured via the database (dynamic configuration), is that it?

---

**summary:** having the configurations that are made via `toml` for the database (a table that allows configuration), is that it?

**Comment by LucasZhanye** [comment]
> Configuration dynamics
Regarding dynamic configuration, for example, in a production environment, when we add a module and need to add a table, we need to control the permissions of this table, but we will not easily restart the prestd service. We prefer to modify it through The configuration will take effect. I checked the code and found that each request goes through the middleware to re-judge the permissions, so dynamic configuration can be satisfied.
> Table permissions
Regarding table permissions, it depends on dynamic configuration. We hope that all tables will be accessible by default. Only tables with configured permissions will be controlled. Like dynamic configuration, we do not want to restart the service, but the current situation is that once it is turned on In addition to permission control, after we add a new table, we need to add the permission configuration of the table, even if all its fields are readable, writable and deleteable.

Therefore, in actual work scenarios, I think it is necessary to support dynamic configuration and control methods for changing table permissions.
